File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-06/plagiarized/L6/04/Kasus6.kt
package IR

import java.util.*

/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/

/**
 *
 * @author C166AB8FBBBE55174E16A67530426A8A 19011F43894F06E2BA0CA74061427469
 */
object Kasus6 {
    fun simpanArray(arr: IntArray, i: Int) {
        var i = i
        val sc = Scanner(System.`in`)

        if (i > 0) {
            simpanArray(arr, --i)
            print("Read a number: ")
            arr[i] = sc.nextInt()
        }
    }

    fun tampilArray(arr: IntArray, i: Int) {
        var i = i
        if (i >= 0) {
            println(arr[i])
            tampilArray(arr, --i)
        }
    }

    @JvmStatic
    fun main(args: Array<String>) {
        var n = 10
        val arr = IntArray(n)
        simpanArray(arr, n)
        tampilArray(arr, --n)
    }
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-06/plagiarized/L6/04/Kasus6.ts
import { Scanner } from 'readline-sync';

function simpanArray(arr: number[], i: number): void {
    const sc = new Scanner();

    if (i > 0) {
        simpanArray(arr, --i);
        console.log('Read a number: ');
        arr[i] = sc.nextInt();
    }
}

function tampilArray(arr: number[], i: number): void {
    if (i >= 0) {
        console.log(arr[i]);
        tampilArray(arr, --i);
    }
}

function main(): void {
    const n = 10;
    const arr: number[] = new Array(n);
    simpanArray(arr, n);
    tampilArray(arr, n - 1);
}

main();

